notebooks/python_intro/05-functions-write.md

```diff
@@ -0,0 +1,283 @@
+
+
+
+Functions are first-class objects, which means they can be:
+
+* assigned to a variable
+* an item in a list (or any collection)
+* passed as an argument to another function.
+
+
+
+```python
+def double_it(x):
+    """Double a value"""
+    return 2*x
+
+def half_it(x):
+    """Divide a value by 2"""
+    return x/2
+
+dit = double_it
+dit(5)
+```
+
+```python
+def apply_it(x, func):
+    return func(x)
+
+apply_it(5, half_it), apply_it(5, double_it)
+```
+
+## Exercise
+
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+   
+The **barometric formula**, sometimes called the exponential atmosphere or isothermal atmosphere, is a formula used to model how the **pressure** (or density) of the air **changes with altitude**. The pressure drops approximately by 11.3 Pa per meter in first 1000 meters above sea level.
+
+$$P=P_0 \cdot \exp \left[\frac{-g \cdot M \cdot h}{R \cdot T}\right]$$
+    
+where:
+* $T$ = standard temperature, 288.15 (K)
+* $R$ = universal gas constant, 8.3144598, (J/mol/K)
+* $g$ = gravitational acceleration, 9.81 (m/s$^2$)
+* $M$ = molar mass of Earth's air, 0.02896 (kg/mol)
+
+and:
+* $P_0$ = sea level pressure (hPa)
+* $h$ = height above sea level (m)    
+    
+The measured air pressure at sea level is 1010 hPa. Write a function `barometric_formula` with two input parameters:
+    
+- `pressure_sea_level`: a mandatory parameter.
+- `height`: a keyword parameter with a default value of 2500.
+    
+Use the function `barometric_formula` to calculate the corresponding air pressure on 2500m and on 2750m above sea level when the measured air pressure at sea level is 1010 hPa.
+
+<details><summary>Hints</summary>
+
+- See notebook 02-functions_use.ipynb for the for the code itsef, but encapsulate this into a function.
+- Do not forget to add a docstring, see the specification on the [numpy docstring documentation](https://numpydoc.readthedocs.io/en/latest/format.html).
+
+FYI, see https://www.math24.net/barometric-formula/ or https://en.wikipedia.org/wiki/Atmospheric_pressure    
+    
+</details>    
+    
+</div>
+
+```python tags=["nbtutor-solution"]
+import math
+
+def barometric_formula(pressure_sea_level, height=2500):
+    """Apply barometric formula
+    
+    Apply the barometric formula to calculate the air pressure on a given height
+    
+    Parameters
+    ----------
+    pressure_sea_level : float
+        pressure, measured as sea level (hPa)
+    height : float
+        height above sea level (m)
+    
+    Notes
+    ------
+    see https://www.math24.net/barometric-formula/ or 
+    https://en.wikipedia.org/wiki/Atmospheric_pressure
+    """
+    standard_temperature = 288.15
+    gas_constant = 8.3144598
+    gravit_acc = 9.81
+    molar_mass_earth = 0.02896
+    
+    pressure_altitude = pressure_sea_level * math.exp(-gravit_acc * molar_mass_earth* height/(gas_constant*standard_temperature))
+    return pressure_altitude
+
+barometric_formula(1010), barometric_formula(1010, 2750)
```

Comment:
As a final step, you could also expand the exercise (or a follow-up exercise) to now use this function for a list of values in a loop or list comprehension

Reply:
great idea!